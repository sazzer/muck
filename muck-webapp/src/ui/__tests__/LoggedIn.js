// @flow

import React from 'react';
import {shallow} from 'enzyme';
import {ConnectedHasAccessToken} from '../LoggedIn';

/** Create the component to test */
function setup({hasAccessToken, children}) {
    const auth = {
        selectHasAccessToken: jest.fn(),

        // These are needed for Flow
        loadAuthenticationServices: jest.fn(),
        selectAccessToken: jest.fn(),
        selectAuthenticationServices: jest.fn(),
        startAuthentication: jest.fn()
    };

    auth.selectHasAccessToken.mockReturnValueOnce(hasAccessToken);

    return {
        auth,
        element: shallow(<ConnectedHasAccessToken auth={auth}>{ children }</ConnectedHasAccessToken>)
    };
}

describe('ConnectedHasAccessToken', () => {
    it('Renders correctly when the access token is missing', () => {
        const { element } = setup({
            hasAccessToken: false,
            children: "Missing"
        });

        expect(element).toMatchSnapshot();
    });
    it('Renders correctly when the access token is present', () => {
        const { element } = setup({
            hasAccessToken: true,
            children: "Present"
        });

        expect(element).toMatchSnapshot();
    });
});
