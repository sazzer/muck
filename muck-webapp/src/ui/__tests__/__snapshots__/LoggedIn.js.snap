// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedHasAccessToken Renders correctly when the access token is missing 1`] = `<MissingAccessToken />`;

exports[`ConnectedHasAccessToken Renders correctly when the access token is present 1`] = `"Present"`;
